src/overlays/actors/ovl_Bg_F40_Swlift/z_bg_f40_swlift.c

```diff
@@ -28,24 +31,83 @@ ActorInit Bg_F40_Swlift_InitVars = {
     (ActorFunc)BgF40Swlift_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_8096F540[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneScale, 550, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 5000, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+void BgF40Swlift_Init(Actor* thisx, PlayState* play) {
+    BgF40Swlift* this = THIS;
+    s32 index;
+    s32 pad;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 1);
+    index = BG_F40_SWLIFT_GET_INDEX(thisx);
+    if ((index < 0) || (index >= 5)) { //! @bug An index greater than 3 will cause an out of bounds array access.
+        Actor_Kill(&this->dyna.actor);
+    } else {
+        sHeights[index] = this->dyna.actor.world.pos.y;
+        sSwitchFlags[index] = BG_F40_SWLIFT_GET_SWITCHFLAG(thisx);
+        if (index != 0) {
+            Actor_Kill(&this->dyna.actor);
+        } else {
+            DynaPolyActor_LoadMesh(play, &this->dyna, &gStoneTowerVerticallyOscillatingPlatformCol);
+            this->dyna.actor.params = 0;
+        }
+    }
+}
+
+void BgF40Swlift_Destroy(Actor* thisx, PlayState* play) {
+    BgF40Swlift* this = THIS;
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void BgF40Swlift_Update(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    BgF40Swlift* this = THIS;
+    s32 i;
 
-extern InitChainEntry D_8096F540[];
+    for (i = 1; i < ARRAY_COUNT(sSwitchFlags); i++) {
+        if ((sSwitchFlags[i] == 0xFF) || !Flags_GetSwitch(play, sSwitchFlags[i])) {
+            break;
+        }
+    }
 
-extern UNK_TYPE D_06003B08;
-extern UNK_TYPE D_06003E80;
+    i--;
+    if (i != this->dyna.actor.params) {
+        f32 heightOffset;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Swlift/BgF40Swlift_Init.s")
+        this->dyna.actor.params = -1;
+        heightOffset = (sHeights[i] - this->dyna.actor.world.pos.y) * 0.1f;
+        if (heightOffset > 0.0f) {
+            heightOffset = CLAMP(heightOffset, 0.5f, 15.0f);
+        } else {
+            heightOffset = CLAMP(heightOffset, -15.0f, -0.5f);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_F40_Swlift/BgF40Swlift_Destroy.s")
+        if ((Math_StepToF(&this->dyna.actor.speed, heightOffset, 1.0f) != 0) && (fabsf(heightOffset) <= 0.5f)) {
```

Comment:
```suggestion
        if (Math_StepToF(&this->dyna.actor.speed, heightOffset, 1.0f) && (fabsf(heightOffset) <= 0.5f)) {
```
`Math_StepToF` returns a boolean